components/camel-snmp/src/test/java/org/apache/camel/component/snmp/TrapTest.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.snmp;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.Message;
+import org.apache.camel.Producer;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.test.AvailablePortFinder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snmp4j.PDU;
+import org.snmp4j.PDUv1;
+import org.snmp4j.mp.SnmpConstants;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.TimeTicks;
+import org.snmp4j.smi.Variable;
+import org.snmp4j.smi.VariableBinding;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class TrapTest extends SnmpTestSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(TrapTest.class);
+
+    @ParameterizedTest
+    @MethodSource("supportedVersions")
+    public void testSendReceiveTraps(int version) throws Exception {
+        PDU trap = createTrap(version);
+
+        // Send it
+        LOG.info("Sending pdu " + trap);
+        Endpoint endpoint = context.getEndpoint("direct:snmptrapV" + version);
+        Exchange exchange = endpoint.createExchange();
+        exchange.getIn().setBody(trap);
+        Producer producer = endpoint.createProducer();
+        producer.process(exchange);
+
+        synchronized (this) {
+            Thread.sleep(1000);
+        }
```

Comment:
Can we do this without the `Thread.sleep`, please? We already have plenty of those making out tests artificially slow.

Reply:
IOW: I think you can use Awaitility here.

Reply:
I'll create another PR with change of this code. TBH I don't remember writing this, and definitely Awaitility should be better here,

Reply:
Thanks @JiriOndrusek, much appreciated!

Reply:
PR: https://github.com/apache/camel/pull/10102